File: Patrones creacionales/abstractfactory/AbstractFactorypy/MiniVan.py
Minivan.py
import abc
class Minivan:
    __metaclass__= abc.ABCMeta
    def __init__(self):
        raise NotImplementedError()
      
    def  getMinivanName(self):
      raise NotImplementedError()
    def getMinivanFeatures(self):
      raise NotImplementedError()
class LuxuryMinivan(Minivan):
    def __init__(self, mName):
      self.__name = mName
    def getMinivanName(self):
       return self.__name
    def getMinivanFeatures(self):
       return "Luxury Mini-Van features"
class NonLuxuryMinivan(Minivan):
  def __init__(self, mName):
     self.__name = mName    
  def getMinivanName(self):
     return self.__name
  def getMinivanFeatures(self):
     return "Non-luxury Mini-Van Fatures"

File: Patrones creacionales/ejercicio puntual/Minivan.java
// Minivan hierarchy
public abstract class Minivan {
    public abstract String getMinivanName();
    public abstract String getMinivanFeatures();
}
class LuxuryMinivan extends Minivan {
    private String name;
    
    public LuxuryMinivan(String mName) {
        this.name = mName;
    }
    @Override
    public String getMinivanName() {
        return this.name;
    }
    @Override
    public String getMinivanFeatures() {
        return "Luxury Mini-Van features";
    }
}
class NonLuxuryMinivan extends Minivan {
    private String name;
    
    public NonLuxuryMinivan(String mName) {
        this.name = mName;
    }
    @Override
    public String getMinivanName() {
        return this.name;
    }
    @Override
    public String getMinivanFeatures() {
        return "Non-luxury Mini-Van Features";
    }
}